source/1_engine/rex_engine/src/diagnostics/logging/internal/common.cpp

```diff
@@ -6,83 +6,74 @@
-  {
-    return m_msg.c_str();
-  }
-
-  REXLOG_INLINE void throw_rexlog_ex(const rex::DebugString& msg, int lastErrno)
-  {
-    REXLOG_THROW(RexlogEx(msg, lastErrno));
-  }
-
-  REXLOG_INLINE void throw_rexlog_ex(const char* msg, int lastErrno)
-  {
-    throw_rexlog_ex(rex::DebugString(msg), lastErrno);
-  }
-
-  REXLOG_INLINE void throw_rexlog_ex(rex::DebugString msg)
-  {
-    REXLOG_THROW(RexlogEx(rsl::move(msg)));
-  }
-
-  REXLOG_INLINE void throw_rexlog_ex(const char* msg)
-  {
-    throw_rexlog_ex(rex::DebugString(msg));
-  }
 
+        static string_view_t g_level_string_views[]
+        {
+            REXLOG_LEVEL_NAME_TRACE, REXLOG_LEVEL_NAME_DEBUG, REXLOG_LEVEL_NAME_INFO, REXLOG_LEVEL_NAME_WARNING, REXLOG_LEVEL_NAME_ERROR, REXLOG_LEVEL_NAME_CRITICAL, REXLOG_LEVEL_NAME_OFF                                                                  \
+        };
+
+        static string_view_t g_short_level_names[]
+        {
+            REXLOG_LEVEL_SNAME_TRACE, REXLOG_LEVEL_SNAME_DEBUG, REXLOG_LEVEL_SNAME_INFO, REXLOG_LEVEL_SNAME_WARNING, REXLOG_LEVEL_SNAME_ERROR, REXLOG_LEVEL_SNAME_CRITICAL, REXLOG_LEVEL_SNAME_OFF                                                           \
+        };
+
+        string_view_t to_string_view(rexlog::level::LevelEnum l) noexcept
+        {
+            switch (l)
+            {
+            case LevelEnum::Off: return REXLOG_LEVEL_NAME_OFF;
+            case LevelEnum::Critical: return REXLOG_LEVEL_NAME_CRITICAL;
+            case LevelEnum::Err: return REXLOG_LEVEL_NAME_ERROR;
+            case LevelEnum::Warn: return REXLOG_LEVEL_NAME_WARNING;
+            case LevelEnum::Info: return REXLOG_LEVEL_NAME_INFO;
+            case LevelEnum::Debug: return REXLOG_LEVEL_NAME_DEBUG;
+            case LevelEnum::Trace: return REXLOG_LEVEL_NAME_TRACE;
+            default:
+                printf("Invalid level::LevelEnum - No conversion possible");
+                return "";
+            }
+        };
+
+        string_view_t to_short_c_str(rexlog::level::LevelEnum l) noexcept
+        {
+            switch (l)
+            {
+            case LevelEnum::Off: return REXLOG_LEVEL_SNAME_OFF;
+            case LevelEnum::Critical: return REXLOG_LEVEL_SNAME_CRITICAL;
+            case LevelEnum::Err: return REXLOG_LEVEL_SNAME_ERROR;
+            case LevelEnum::Warn: return REXLOG_LEVEL_SNAME_WARNING;
+            case LevelEnum::Info: return REXLOG_LEVEL_SNAME_INFO;
+            case LevelEnum::Debug: return REXLOG_LEVEL_SNAME_DEBUG;
+            case LevelEnum::Trace: return REXLOG_LEVEL_SNAME_TRACE;
+            default:
+                printf("Invalid level::LevelEnum - No conversion possible");
+                return "";
+            }
+        }
+
+        rexlog::level::LevelEnum from_str(const rex::DebugString& name) noexcept
+        {
+            const auto* it = rsl::find(rsl::begin(g_level_string_views), rsl::end(g_level_string_views), name);
```

Comment:
I find it generally better to use cbegin, this presents to the reader you're not gonna change anything underneath